app/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSenderTest.groovy

```diff
@@ -1,14 +1,97 @@
 package gov.hhs.cdc.trustedintermediary.external.reportstream
 
+import gov.hhs.cdc.trustedintermediary.context.TestApplicationContext
 import gov.hhs.cdc.trustedintermediary.etor.demographics.LabOrder
+import gov.hhs.cdc.trustedintermediary.etor.demographics.LabOrderSender
+import gov.hhs.cdc.trustedintermediary.external.jackson.Jackson
+import gov.hhs.cdc.trustedintermediary.external.slf4j.Slf4jLogger
+import gov.hhs.cdc.trustedintermediary.wrappers.AuthEngine
+import gov.hhs.cdc.trustedintermediary.wrappers.Formatter
+import gov.hhs.cdc.trustedintermediary.wrappers.HapiFhir
+import gov.hhs.cdc.trustedintermediary.wrappers.HttpClient
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger
 import spock.lang.Specification
 
 class ReportStreamLabOrderSenderTest extends Specification{
 
+    def setup() {
+        TestApplicationContext.reset()
+        TestApplicationContext.init()
+        TestApplicationContext.register(LabOrderSender, ReportStreamLabOrderSender.getInstance())
+    }
+
+    def "sendRequestBody works"() {
+        given:
+        def mockClient = Mock(HttpClient)
+        TestApplicationContext.register(HttpClient, mockClient)
+        TestApplicationContext.injectRegisteredImplementations()
+
+        when:
+        TestApplicationContext.getImplementation(LabOrderSender).sendRequestBody("message_1", "fake token")
+        TestApplicationContext.getImplementation(LabOrderSender).sendRequestBody("message_2", "fake token")
+
+        then:
+        2 * mockClient.post(_ as String, _ as Map<String,String>, _ as String) >> "200"
+    }
+
+    def "requestToken works"() {
+        given:
+        def expected = "rs fake token"
+        def mockAuthEngine = Mock(AuthEngine)
+        def mockClient = Mock(HttpClient)
+        TestApplicationContext.register(AuthEngine, mockAuthEngine)
+        TestApplicationContext.register(HttpClient, mockClient)
+        TestApplicationContext.register(Logger, Slf4jLogger.getLogger())
+        TestApplicationContext.register(Formatter, Jackson.getInstance())
+        TestApplicationContext.injectRegisteredImplementations()
+        when:
+        def actual = ReportStreamLabOrderSender.getInstance().requestToken()
+        then:
+        1 * mockAuthEngine.generateSenderToken(_ as String, _ as String, _ as String, _ as String, 300) >> "sender fake token"
+        1 * mockClient.requestToken(_ as String, _ as Map<String,String>, _ as String) >> """{"access_token":"rs fake token", "token_type":"bearer"}"""
```

Comment:
You can inject the expected variable into the JSON blob to reduce the fragility of the test.

```suggestion
        1 * mockClient.requestToken(_ as String, _ as Map<String,String>, _ as String) >> """{"access_token":"${expected}", "token_type":"bearer"}"""
```